src/network/evm_chain.py

```diff
@@ -19,19 +38,31 @@ def __init__(
         tracked_contracts=[],
         oracle_metadata=config.ORACLE_CONTRACT_ABI,
     ):
-        super().__init__(name, url, credentials, active, tracked_contracts)
-
+        self.chain_data = fetch_chain_data(name)
         self.oracle_metadata = oracle_metadata
+        super().__init__(name, url, credentials, active, tracked_contracts)
 
     def get_connection(self) -> Web3:
-        """ Return Web3 connection to the chain specified by `url`"""
+        """Return Web3 connection to the chain specified by `url`."""
         if self.url.startswith("ws"):
-            return Web3(Web3.WebsocketProvider(self.url))
+            web3 = Web3(Web3.WebsocketProvider(self.url))
         elif self.url.startswith("http"):
-            return Web3(Web3.HTTPProvider(self.url))
+            web3 = Web3(Web3.HTTPProvider(self.url))
         else:
             raise ValueError("URL type not supported")
 
+        if self.chain_data and (
```

Comment:
Since `ChainValidationFailed` is a runtime exception and not a RPC exception, it would make sense for this to be checked on the initialization of the chain. When we introduce connection pools, this check should be moved there. Or potentially even in `Chains` class. Not sure though. (just a comment)

Reply:
Good call, I've introduced a `validate_chain` method within `EvmChain` which is called on initialisation.  As you say it will most likely be better placed in the connection pool once it is introduced.